src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBrokerHelper.cs

```diff
@@ -173,29 +172,44 @@ public void GetBrokerAccountInfo(IDictionary<string, string> brokerPayload, Acti
 
             if (result == null)
             {
-                _logger.Info("Android account manager AddAccount didn't return any accounts. ");
-                throw new MsalException(MsalError.NoAndroidBrokerAccountFound, "Please add the selected account to the broker");
+                _logger.Info("Android account manager didn't return any accounts.");
+                throw new MsalUiRequiredException(MsalError.NoAndroidBrokerAccountFound, MsalErrorMessage.NoAndroidBrokerAccountFound);
             }
 
             Bundle bundleResult = (Bundle)result?.Result;
             var accounts = bundleResult?.GetString(BrokerConstants.BrokerAccounts);
 
+            string username = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.Username);
+            string homeAccountId = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.HomeAccountId);
+            string localAccountId = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.LocalAccountId);
+
             if (!string.IsNullOrEmpty(accounts))
             {
                 dynamic authResult = JArray.Parse(accounts);
 
                 foreach (JObject account in authResult)
                 {
                     var accountData = account[BrokerResponseConst.Account];
-                    if ((accountData[BrokerResponseConst.UserName]).ToString() == loginHint)
+
+                    if ((accountData[BrokerResponseConst.UserName]).ToString() == username)
```

Comment:
Do not compare strings like this. Use string.Equals please.

Reply:
Why? '==' operator calls Equals to do ordinal comparison anyway. Unless there are consequences of Xamarin translating it differently.